### PR TITLE
Faster constraint check

### LINKED ISSUE #363: Exclude hydrogens from being mapped
In order to allow us to run `perses` with bonds to hydrogen constrained, we are forced to exclude hydrogens from mapped atoms so that they are always `unique`. Otherwise, we would not be able to permit constrained `H-X` bonds of different lengths in initial and final states.

### PR DESCRIPTION
**Pull request by pgrinaway**
This PR updates the constraint check to use the system XMLs and lxml; this is much faster (on the order of 15sec instead of 15min) and allows us to re-introduce the constraint check. This should resolve #363 by raising an exception if a constraint is changing.

### CODE REVIEW

File: perses/annihilation/new_relative.py [hunk 1]
```diff
@@ -410,6 +412,50 @@ def _constraint_check(self):
...
+        old_system_tree = etree.parse(old_system_io)
+        new_system_tree = etree.parse(new_system_io)
+
+        #get the list of constraints from new and old systems:
+        old_system_constraint_list = old_system_tree.xpath(constraint_string)
+        new_system_constraint_list = new_system_tree.xpath(constraint_string)
+
+        #convert the list of constraint elements to dictionaries. By using frozenset, we can do this independent of the order of
+        old_system_constraints = {frozenset((o_h_map[int(constraint.attrib['p1'])], o_h_map[int(constraint.attrib['p2'])])) : float(constraint.attrib['d']) for constraint in old_system_constraint_list}
+        new_system_constraints = {frozenset((n_h_map[int(constraint.attrib['p1'])], n_h_map[int(constraint.attrib['p2'])])) : float(constraint.attrib['d']) for constraint in new_system_constraint_list}
```
Comment (andrrizzi) [inline perses/annihilation/new_relative.py]: I think it should be safe to use simply a `tuple` instead of a `frozenset` for the keys.

Comment (andrrizzi) [reply perses/annihilation/new_relative.py]: Ah! Or is it to avoid the case when you have duplicate `(p1, p2)` and `(p2, p1)` pairs?

Comment (pgrinaway) [reply perses/annihilation/new_relative.py]: But if I do that, then the order of the indices will matter, right?

Comment (pgrinaway) [reply perses/annihilation/new_relative.py]: Oops, didn't see your other message.

**Pull request merged**